tests/frameworks/OSXShims/src/LoggingNative.mm

```diff
@@ -0,0 +1,60 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <LoggingNative.h>
+#import <Foundation/NSString.h>
+
+#import <cstdarg>
+
+void TraceVerbose(const wchar_t* tag, const wchar_t* format, ...) {
+    NSString* formatString = [NSString stringWithFormat:@"[%ws] %ws", tag, format];
+    va_list varArgs;
+    va_start(varArgs, format);
+    NSLog(formatString, varArgs);
```

Comment:
[https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Miscellaneous/Foundation_Functions/#//apple_ref/c/func/NSLog](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Miscellaneous/Foundation_Functions/#//apple_ref/c/func/NSLog)

void NSLog ( NSString *format, ... );

Simply calls NSLogv, passing it a variable number of arguments.

---
In reply to: [72312621](https://github.com/Microsoft/WinObjC/pull/634#discussion_r72312621) [](ancestors = 72312621)